gateway/src/shard/processor/socket_forwarder.rs

```diff
@@ -41,18 +37,12 @@ impl SocketForwarder {
         #[cfg(feature = "tracing")]
         tracing::debug!("starting driving loop");
 
-        loop {
-            let timeout = sleep(Self::TIMEOUT).fuse();
-            tokio::pin!(timeout);
-
-            let rx = Box::pin(self.rx.recv().fuse());
-            let tx = Box::pin(self.stream.next().fuse());
+        let timeout = sleep(Self::TIMEOUT);
+        tokio::pin!(timeout);
 
-            let select_message = future::select(rx, tx);
-
-            match future::select(select_message, timeout).await {
-                // `rx` future finished first.
```

Comment:
Are these comments still relevant? could they be retained if so?

Reply:
I think they were there to help disambiguate `Either::Left((Either::Left((maybe_msg, _)), _))`, not which finished "first", `tokio::select!` is self documenting so I don't feel it necessary to keep them.